lib/internal/modules/esm/get_format.js

```diff
@@ -41,6 +45,12 @@ function getDataProtocolModuleFormat(parsed) {
   return mimeToFormat(mime);
 }
 
+/**
+ * @param {URL} url
+ * @param {{parentURL: string}} context
```

Comment:
Nit: `context` exists in a bunch of places, so it would be better to create a typedef and then reference that. I'd be happy to do in a follow-up too.

Reply:
Since it's Bradley code, I feel inclined to land it as is (if I can get at least another 👍  on https://github.com/nodejs/node/pull/42305#issuecomment-1065664749), and let you make a follow up PR since this one as already several approvals.

Reply:
Along these lines, and also to consider in the follow-up PR rather than here, shouldn’t we be using true JSDoc syntax rather than TypeScript inside brackets? In other words, instead of:

```js
 * @param {{parentURL: string}} context
```

We would write:

```js
 * @param {object} context
 * @param {string} context.parentURL
```

Besides being correct JSDoc (in my understanding) the latter lets us add descriptions to each property of the object, like `parentURL` in this case.